jolt-core/src/poly/commitment/dory.rs

```diff
@@ -1337,6 +1338,128 @@ impl CommitmentScheme for DoryCommitmentScheme {
+                .expect("MSM calculation failed."),
+        );
+        vec![row_commitment]
+    }
+
+    fn process_chunk_onehot<'a>(
+        state: &Self::State<'a>,
+        chunk: &[Option<usize>],
+    ) -> Self::ChunkState {
+        let Some(K) = state.K else {
+            panic!("K must be provided for OneHot polynomials.");
+        };
+
+        let mut indices_per_k: Vec<Vec<usize>> = vec![Vec::new(); K];
+        for (col_index, k) in chunk.iter().enumerate() {
+            // All the nonzero coefficients are 1, so we simply add
+            // the associated base to the result.
+            if let Some(k) = k {
+                indices_per_k[*k].push(col_index);
+            }
+        }
+
+        let results = jolt_optimizations::batch_g1_additions_multi(state.bases, &indices_per_k);
+
+        let mut row_commitments = vec![JoltGroupWrapper(<Bn254 as ArkPairing>::G1::zero()); K];
+        for (k, result) in results.into_iter().enumerate() {
+            if !indices_per_k[k].is_empty() {
+                let sum_projective: G1Projective = result.into();
+                row_commitments[k].0 = sum_projective;
+            }
+        }
+        row_commitments
+    }
+
+    fn finalize<'a>(
+        state: &Self::State<'a>,
+        chunks: &[Self::ChunkState],
+    ) -> (Self::Commitment, Self::OpeningProofHint) {
+        if let Some(K) = state.K {
+            let row_len = DoryGlobals::get_num_columns();
+            let T = DoryGlobals::get_T();
+            let rows_per_k = T / row_len;
+            let num_rows = K * T / row_len;
+
+            let mut row_commitments = vec![JoltGroupWrapper(G1Projective::zero()); num_rows];
+            for (chunk_index, commitments) in chunks.iter().enumerate() {
+                row_commitments
+                    .par_iter_mut()
+                    .skip(chunk_index)
+                    .step_by(rows_per_k)
+                    .zip(commitments.into_par_iter())
+                    .for_each(|(dest, src)| *dest = *src);
+            }
+
+            let commitment = JoltBn254::multi_pair(
+                &row_commitments,
+                &state.setup.g2_vec()[..row_commitments.len()],
+            );
+            (DoryCommitment(commitment), row_commitments)
+        } else {
+            let T = DoryGlobals::get_T();
+            let mut row_commitments: Vec<_> = chunks.into_par_iter().map(|r| r[0]).collect();
+            // Pad row commitments since we don't pad streamed trace.
+            let row_pad_count = T / DoryGlobals::get_num_columns() - row_commitments.len();
+            assert!(T / DoryGlobals::get_num_columns() >= row_commitments.len());
+            row_commitments.extend(vec![JoltG1Wrapper::identity(); row_pad_count]);
+            let commitment = JoltBn254::multi_pair(
+                &row_commitments,
+                &state.setup.g2_vec()[..row_commitments.len()],
+            );
```

Comment:
I don't think we need to pad here?
```suggestion
            let commitment = JoltBn254::multi_pair(
                &row_commitments,
                &state.setup.g2_vec()[..row_commitments.len()],
            );
```